### PR TITLE
Pass state.isShadow to custom nodeCanvasObject

### PR DESCRIPTION
**Pull request by stefanprobst**
A custom `nodeCanvasObject` currently does not receive `isShadow` which would be helpful to fully replicate the default node drawing behavior.

### CODE REVIEW

**Comment by vasturiano** [comment]
For my understanding, why is it important in your use case to know whether it's painting on the main or the shadow canvas?

**Comment by vasturiano** [comment]
Ok, makes sense, thanks for the explanation. Would you mind also updating the README?

**Comment by stefanprobst** [comment]
Done :+1:

File: README.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ myGraph(<myDOMElement>)
...
 | <b>nodeAutoColorBy</b>([<i>str</i> or <i>fn</i>]) | Node object accessor function (`fn(node)`) or attribute (e.g. `'type'`) to automatically group colors by. Only affects nodes without a color attribute. | |
-| <b>nodeCanvasObject</b>([<i>fn</i>]) | Callback function for painting a custom canvas object to represent graph nodes. Should use the provided canvas context attribute to perform drawing operations for each node. The callback function will be called for each node at every frame, and has the signature: `.nodeCanvasObject(<node>, <canvas context>, <current global scale>)`.  | *default node object is a circle, sized according to `val` and styled according to `color`.* |
+| <b>nodeCanvasObject</b>([<i>fn</i>]) | Callback function for painting a custom canvas object to represent graph nodes. Should use the provided canvas context attribute to perform drawing operations for each node. The callback function will be called for each node at every frame, and has the signature: `.nodeCanvasObject(<node>, <canvas context>, <current global scale>), <isShadowContext>`.  | *default node object is a circle, sized according to `val` and styled according to `color`.* |
```
Comment (vasturiano) [inline README.md]: Typo on parenthesis placement.

**Comment by vasturiano** [comment]
Thanks @stefanprobst!

**Pull request merged**